### PR TITLE
Add docstrings for PhyloXML

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes a part of #1203 

There are a lot of places where docstrings are missing, so the PR aims to start a discussion on adding the docstrings to all methods and classes.

### CODE REVIEW

**Comment by peterjc** [comment]
I'll leave reviewing this to @etal - note there's a minor white space thing from the flake docstring checks,

```
$ flake8 Bio/
Bio/Phylo/PhyloXML.py:589:1: D413 Missing blank line after last section
Bio/Phylo/PhyloXML.py:600:1: D413 Missing blank line after last section
```

File: Bio/Phylo/PhyloXML.py [hunk 1]
```diff
@@ -593,6 +593,7 @@ def __eq__(self, other):
             value : float
                 confidence value
         """
+
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: That should not be a blank line according to ``pydocstyle``, this breaks its rule ``D202 No blank lines allowed after function docstring``

File: Bio/Phylo/PhyloXML.py [hunk 2]
```diff
@@ -604,6 +605,7 @@ def __ne__(self, other):
         value : float
             confidence value.
         """
+
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: That should not be a blank line according to ``pydocstyle``, this breaks its rule ``D202 No blank lines allowed after function docstring``

**Comment by peterjc** [comment]
You didn't put those two 'blank' lines where ``pydocstyle`` (via ``flake8``) expected. It mean inside the docstring.

```
$ flake8 Bio/
Bio/Phylo/PhyloXML.py:589:1: D202 No blank lines allowed after function docstring
Bio/Phylo/PhyloXML.py:589:1: D413 Missing blank line after last section
Bio/Phylo/PhyloXML.py:596:1: W293 blank line contains whitespace
Bio/Phylo/PhyloXML.py:601:1: D202 No blank lines allowed after function docstring
Bio/Phylo/PhyloXML.py:601:1: D413 Missing blank line after last section
Bio/Phylo/PhyloXML.py:608:1: W293 blank line contains whitespace
```

**Comment by souravsingh** [comment]
Ping @talevich

**Comment by etal** [comment]
Looks good to me now, thanks!

**Pull request merged**